internal/statestore/backfill_test.go

```diff
@@ -301,3 +302,54 @@ func TestDeleteBackfill(t *testing.T) {
 	require.Equal(t, codes.Unavailable.String(), status.Convert(err).Code().String())
 	require.Contains(t, status.Convert(err).Message(), "DeleteBackfill, id: 12345, failed to connect to redis:")
 }
+
+func TestAcknowledgeBackfill(t *testing.T) {
```

Comment:
Actually, it's not an _AcknowledgeBackfill_ unit test now. You test here the whole acknowledge scenario. This is nice, but I would call this test something like _TestAcknowledgeLifecycle_. 
To write a unit test you need to test exactly one single unit (method) under the different conditions. 
Talking about backfill it's enough to check that after calling _AcknowledgeBackfill_ the passed ID is added to _backfill_last_ack_time_.
So I suggest you to rename this method and add _TestAcknowledgeBackfill_ where you will test only _AcknowledgeBackfill_ method. What do you think?